pylearn2/devtools/nan_guard.py

```diff
@@ -36,6 +37,23 @@ class NanGuardMode(Mode):
         If True, raise an error when a value greater than 1e10 is encountered.
     """
     def __init__(self, nan_is_error, inf_is_error, big_is_error=True):
+        self.guard_input = T.vector('nan_guard')
```

Comment:
This force the transfer of the data to the from the GPU to the CPU, then to the GPU again. You should declare this as theano.sandbox.cuda.fvector()